freqtrade/persistence/models.py

```diff
@@ -403,31 +459,63 @@ def adjust_stop_loss(self, current_price: float, stoploss: float,
             f"Trailing stoploss saved us: "
             f"{float(self.stop_loss) - float(self.initial_stop_loss):.8f}.")
 
+    def is_opening_trade(self, side) -> bool:
+        """
+        Determines if the trade is an opening (long buy or short sell) trade
+        :param side (string): the side (buy/sell) that order happens on
+        """
+        return (side == 'buy' and not self.is_short) or (side == 'sell' and self.is_short)
+
+    def is_closing_trade(self, side) -> bool:
+        """
+        Determines if the trade is an closing (long sell or short buy) trade
+        :param side (string): the side (buy/sell) that order happens on
+        """
+        return (side == 'sell' and not self.is_short) or (side == 'buy' and self.is_short)
+
     def update(self, order: Dict) -> None:
         """
         Updates this entity with amount and actual open/close rates.
         :param order: order retrieved by exchange.fetch_order()
         :return: None
         """
         order_type = order['type']
+
+        if ('leverage' in order and 'borrowed' in order):
+            raise OperationalException(
+                'Pass only one of Leverage or Borrowed to the order in update trade')
+
+        if 'is_short' in order and order['side'] == 'sell':
+            # Only set's is_short on opening trades, ignores non-shorts
+            # TODO-mg: I don't like this, but it might be the only way
+            self.is_short = order['is_short']
+
```

Comment:
I doubt this will work in practice (it might in tests  ... but see comment on the test fixture).

the order object does not contain a "is_short" property - at least not "formalized / unionized by ccxt.
Sure, we MIGHT be able to inject it in the exchange class ... but i'm not sure that'll work correctly...